pylzy/lzy/env/explorer/classify.py

```diff
@@ -199,6 +212,105 @@ def _classify_modules_without_distributions(
 
         return result
 
+    def _classify_meta_package_distributions(
+        self,
+        packages: Set[BasePackage],
+        binary_distributions: DistributionSet,
+    ) -> Set[BasePackage]:
+        """
+        Here we are trying to find and classify meta packages.
+        We are considering as meta package any package that doesn't contain any files
+        except for its meta.
+        Problem is, these meta-packages can't be found by usual module exploring because there no
+        modules in it.
+
+        For example, meta package "tensorflow" requires usual package "tensorflow-intel".
+        We must find "tensorflow" in our exploring process to send it as requirement to server,
+        because server could have different platform from local machine and requirements
+        from meta package "tensorflow" will be compiled to another packages list.
+        (Also we will filter "tensorflow-intel" from requirements because server have
+        wrong platorm for it, look at `_check_distribution_platform_at_pypi`)
+
+        """
+
+        result: Set[BasePackage] = set()
+        meta_packages: DistributionSet = set()
+        seen_meta_packages: Set[str] = set()
+
+        def add_meta_requirements(name: str) -> None:
+            """here we check if package with 'name' are required by one or more
+            meta-packages and if it is, we adding these meta packages to our DFS"""
+
+            new_meta_packages = self.requirements_to_meta_packages.get(name, ())
+            for meta_package in new_meta_packages:
+                if meta_package.name not in seen_meta_packages:
+                    meta_packages.add(meta_package)
```

Comment:
since `meta_packages` is a set, why we need to support `seen_meta_packages` instead of just `add()`?

Reply:
Because we could get an early popped meta package later on, it will lead to an infinite loop.